pkg/controllers/user/rbac/cluster_handler.go

```diff
@@ -21,54 +24,82 @@ func newClusterHandler(workload *config.UserContext) v3.ClusterHandlerFunc { //*
 	informer.AddIndexers(indexers)
 
 	ch := &clusterHandler{
-		grbIndexer:    informer.GetIndexer(),
+		grbIndexer: informer.GetIndexer(),
+		// Management level resources
 		grbController: workload.Management.Management.GlobalRoleBindings("").Controller(),
 		clusters:      workload.Management.Management.Clusters(""),
-		clusterName:   workload.ClusterName,
+		// User context resources
+		userGRB:       workload.RBAC.ClusterRoleBindings(""),
+		userGRBLister: workload.RBAC.ClusterRoleBindings("").Controller().Lister(),
 	}
 	return ch.sync
 }
 
 type clusterHandler struct {
+	grbIndexer cache.Indexer
+	// Management level resources
 	grbController v3.GlobalRoleBindingController
-	grbIndexer    cache.Indexer
 	clusters      v3.ClusterInterface
-	clusterName   string
+	// User context resources
+	userGRB       v1.ClusterRoleBindingInterface
+	userGRBLister v1.ClusterRoleBindingLister
 }
 
-func (h *clusterHandler) sync(key string, cluster *v3.Cluster) (runtime.Object, error) {
-	if key == "" || cluster == nil || cluster.Name != h.clusterName {
+func (h *clusterHandler) sync(key string, obj *v3.Cluster) (runtime.Object, error) {
+	if key == "" || obj == nil {
 		return nil, nil
 	}
 
-	original := cluster
-	cluster = original.DeepCopy()
-
-	var updateErr error
-	err := h.doSync(cluster)
-	if cluster != nil && !reflect.DeepEqual(cluster, original) {
-		_, updateErr = h.clusters.Update(cluster)
-	}
-
-	if err != nil {
-		return nil, err
+	if !v3.ClusterConditionGlobalAdminsSynced.IsTrue(obj) {
+		err := h.doSync(obj)
+		if err != nil {
+			return nil, err
+		}
+		return h.clusters.Update(obj)
```

Comment:
do you have to do this update? I don't see where the cluster is modified

Reply:
Couple lines down